### PR TITLE
Use testerReady call via config server for tester containers if flag …

### PR DESCRIPTION
**Pull request by hmusum**
…is set

### CODE REVIEW

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/InternalStepRunner.java [hunk 1]
```diff
@@ -455,6 +456,22 @@ private boolean containersAreUp(ApplicationId id, ZoneId zoneId, DualLogger logg
         return true;
     }
 
+    /** Returns true iff all containers in the tester deployment give 100 consecutive 200 OK responses on /status.html. */
+    private boolean testerContainersAreUp(ApplicationId id, ZoneId zoneId, DualLogger logger) {
+        if (useConfigServerForTesterAPI(zoneId)) {
+            DeploymentId deploymentId = new DeploymentId(id, zoneId);
+            if (controller.jobController().cloud().testerReady(deploymentId)) {
```
Comment (jonmv) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/InternalStepRunner.java]: Should this use the config server way as well?

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/InternalStepRunner.java [hunk 2]
```diff
@@ -525,38 +542,58 @@ private String humanize(Node.ServiceState state) {
...
+                                                        controller.applications().contentClustersByZone(deployments));
+        if (useConfigServerForTesterAPI(zoneId)) {
+            controller.serviceRegistry().configServer().startTests(getTesterDeploymentId(id), suite, config);
+        } else {
+            controller.jobController().cloud().startTests(testerEndpoint.get(), suite, config);
+        }
         return Optional.of(running);
     }
 
+    private boolean testerReady(RunId id, URI testerEndpoint) {
```
Comment (jonmv) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/InternalStepRunner.java]: Was this used? (Reviewing on phone...)

**Review by jonmv — APPROVED** [review]
Please look at comments:)

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/InternalStepRunner.java (hunk 1 above)
Comment (hmusum) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/InternalStepRunner.java]: It does (if argument is deployment id the config server will be used)

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/InternalStepRunner.java (hunk 2 above)
Comment (hmusum) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/InternalStepRunner.java]: Oh, forgot to remove that one (ended up not being able to use it after all), will remove in next PR

**Pull request merged**